go/arrow/array/binary.go

```diff
@@ -117,6 +118,21 @@ func (a *Binary) setData(data *Data) {
 	}
 }
 
+func (a *Binary) getOneForMarshal(i int) interface{} {
+	if a.IsNull(i) {
+		return nil
+	}
+	return a.Value(i)
+}
+
+func (a *Binary) MarshalJSON() ([]byte, error) {
+	vals := make([]interface{}, a.Len())
+	for i := 0; i < a.Len(); i++ {
+		vals[i] = a.getOneForMarshal(i)
+	}
+	return json.Marshal(vals)
```

Comment:
How do non-printable characters get marshalled in json?  Escaped bytes?

Reply:
Okay, based on below is looks like base64 encoded?  This is not something I could find as a reference in the docs?

Reply:
From the Go docs on JSON marshalling: https://pkg.go.dev/encoding/json#Marshal

> Array and slice values encode as JSON arrays, except that []byte encodes as a base64-encoded string, and a nil slice encodes as the null JSON value.

The standard marshalling of byte slices to JSON in go will marshal it to a base64-encoded string.